src/bp/core/services/bot-service.ts

```diff
@@ -291,10 +300,18 @@ export class BotService {
   }
 
   private async _executeStageChangeHooks(beforeRequestConfig: BotConfig, currentConfig: BotConfig) {
+    const workspaceId = await this.workspaceService.getBotWorkspaceId(currentConfig.id)
+    const pipeline = await this.workspaceService.getPipeline(workspaceId)
+    if (!pipeline) {
+      return
+    }
+
     const bpConfig = await this.configProvider.getBotpressConfig()
     const alteredBot = _.cloneDeep(currentConfig)
-    const users = await this.workspaceService.listUsers(['email', 'role'])
-    const pipeline = await this.workspaceService.getPipeline()
+
+    // TODO
```

Comment:
? what is todo here

Reply:
Good catch. I forgot to implement that part